src/exprestive.coffee

```diff
@@ -25,111 +23,43 @@ class Exprestive
 
     # The directory where routes and controllers can be found
     # Used only as a relative directory for @routesFilePath and @controllersDirPath
-    @appDir = path.resolve baseDir, @options.appDir
+    appDir = path.resolve baseDir, @options.appDir
 
     # Path to the routes file
-    @routesFilePath = path.resolve @appDir, @options.routesFilePath
+    routesFilePath = path.resolve appDir, @options.routesFilePath
 
     # Path to the directory to look for controllers
-    @controllersDirPath = path.resolve @appDir, @options.controllersDirPath
+    controllersDirPath = path.resolve appDir, @options.controllersDirPath
 
     # Router to register routes and pass to express as middleware
     @middlewareRouter = express.Router()
 
-    # Object to store controllers
-    @controllers = new ControllerStore @controllersDirPath, @options.dependencies
-
     # Save reverse routes
     @reverseRoutes = @options.routes
-    @middlewareRouter.use @setReverseRoutesOnReqLocals
+
+    # Initialize controllers and routes
+    @controllers = new ControllerInitializer controllersDirPath, @options.dependencies
+    {routes} = new RoutesInitializer routesFilePath, @reverseRoutes
```

Comment:
It's a little unusual that an object instance is a hash here, or you use only a part of the instance.